js/stores/windowStore.js

```diff
@@ -750,6 +752,22 @@ const doAction = (action) => {
         }
         break
       }
+    case windowConstants.WINDOW_ON_TOGGLE_BOOKMARK:
+      {
+        // show the AddEditBookmarkHanger control; saving/deleting takes place there
+        const activeFrame = frameStateUtil.getActiveFrame(windowState)
+        let siteDetail = siteUtil.getDetailFromFrame(activeFrame, siteTags.BOOKMARK)
+        const key = siteUtil.getSiteKey(siteDetail)
+
+        if (key !== null) {
```

Comment:
the data it gets from `appStore` is the site which matches... `getDetailFromFrame` should be returning the entry from the sites map (which is what `siteUtil` does; it's the wrapper around the sites map which is stored in `appStore`)

In this case, `getDetailFromFrame` is just taking data from the frame and returning it... rather than doing a lookup and fetching the already existing one.  This code here then copies the data from the existing one (which is a mistake IMO, since it could have just used the existing reference)